lib/fast_stats/mean.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+# A class for managing the arithmetic and geometric means
+# of positive numbers
+module FastStats
+  class Mean
+
+    DEFAULT_ROUND = 15
+    attr_reader :name, :sum, :log_sum, :n
+
+    def initialize(name:)
+      @name = name
```

Comment:
I see how it's important to provide a name when this is being used by `FastStats::Means` so you can differentiate between your various means, but it doesn't seem particularly relevant or useful when dealing with a single `FastStats::Mean`.

Is it possible to make this an optional value? That would change the `summary` method, but only slightly. I might appreciate being able to use this more like
```ruby
mean = FastStats::Mean.new
10.times { |i| mean << i }

puts mean.summary
# => {
#   "arithmetic"=>4.5,
#   "geometric"=>3.597297064377001
# }
```
and then have `FastStats::Means` produce more deeply nested values
```ruby
means.summary
#=> {
#   "foo"=>{
#      "arithmetic"=>4.89,
#      "geometric"=>3.35
#   },
#   "bar"=>{
#      "arithmetic"=>13.64,
#      "geometric"=>7.18
#   }
#}
```
which I imagine would make a summary easier to iterate over.

Reply:
Yes. I like this; particularly the part about producing the more deeply nested values.

Reply:
Oh that is a great suggestion! Yes I like that as well. Ty!